src/custom-gutter-component.coffee

```diff
@@ -0,0 +1,107 @@
+{setDimensionsAndBackground} = require './gutter-component-helpers'
+
+# This class represents a gutter other than the 'line-numbers' gutter.
+# The contents of this gutter may be specified by Decorations.
+
+module.exports =
+class CustomGutterComponent
```

Comment:
:art: Since we now have a `LineNumberGutterComponent`, what do you think about dropping `Custom` and simply name this `GutterComponent`? I feel like this would be consistent with the other names adopted here as well, e.g. we have `Gutter` as opposed to `CustomGutter`. 

I didn't follow this PR since its beginning, therefore, please, excuse me I am bringing up a point you already discussed :pray: